csharp_dotnetcore/14.NLP-With-Dispatch/NlpDispatch/BotServices.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights;
+using Microsoft.Bot.Builder.AI.Luis;
+using Microsoft.Bot.Builder.AI.QnA;
+
+namespace NLP_With_Dispatch_Bot
+{
+    /// <summary>
+    /// Represents the bot's references to external services.
+    ///
+    /// For example, Application Insights, Luis models and QnaMaker services
+    /// are kept here (singletons).  These external services are configured
+    /// using the BotConfigure class (based on the contents of your ".bot" file).
+    /// </summary>
+    [Serializable]
```

Comment:
>[Serializable] [](start = 4, length = 14)

A plague upon this! :) 

No need for Serializable, at least not that I know of. This is old-school dark magic. We enlightened folks now use JSON and other serialization techniques.